swarm/network/stream/delivery.go

```diff
@@ -149,10 +152,10 @@ func (d *Delivery) handleRetrieveRequestMsg(sp *Peer, req *RetrieveRequestMsg) e
 			select {
 			case <-chunk.ReqC:
 			case <-t.C:
-				chunk.SetErrored(true)
+				chunk.SetErrored(storage.ChunkErrTimeout)
 				return
 			}
-			chunk.SetErrored(false)
+			chunk.SetErrored(storage.ChunkErrOk)
```

Comment:
This is confusing. This is not an error at all.

I am starting to hate with passion the fact that a chunk also has the responsibility of a request, and because of this can fall into multiple states. cc @janos @zelig I think we should be changing this asap.

Reply:
👍

Reply:
Can rename to `SetErrorState`?

Reply:
I am not sure how you can make it better. You could change it to `SetState` (because not all `States` are errors), but this won't make it much better. The underlying problem is that `chunk` is not just a data chunk of 4KB that we store in memstore and persist on disk, but also a `request for chunk` which can take multiple states.

Probably `SetErrored` shouldn't have been introduced in the first place.

Reply:
It was not a suggestion to fix the scope of responsibility a chunk has. It was merely for sake of clarity.

Reply:
I know this is out-of-scope of this PR, but we keep on adding small bits of functionality to already bad abstractions, and I just want to raise that. Rather than maintaining the status quo, I want us to have a plan when/if we are going to address these things.

If we are speaking strictly about this particular PR, then I suggest changing the method to `SetState` and renaming `ChunkErrOk` to `ChunkRequestOk` or smth.

Reply:
I can do it, sure. But I think generally it's better if we don't hold up PRs with such out-of-context stuff, and instead just create new issues; I would have been happy to create a new PR straight way and fix this after merge.

Reply:
Yes, sorry about polluting the PR with so many comments, it gets overwhelming quickly and is not the right place for these discussions.